### ISSUE TITLE
Logical early exit should be considered

### ISSUE THREAD

**Issue by GoogleCodeExporter** [typetools/checker-framework#196]
```
What steps will reproduce the problem?

@Nullable Object a;
if (a == null || a.doSomthing())
...

What is the expected output? What do you see instead?
shouldn't throw an error


What version of the product are you using?
1.4.3
```

Original issue reported on code.google.com by `DannyNul...@gmail.com` on 20 Jan 2013 at 3:48
